File: frontend/Models/Quotation.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace frontend.Models
{
    public class QuotationItem
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("quotation_id")]
        public Guid QuotationId { get; set; }

        [JsonPropertyName("item_norma")]
        public string? ItemNorma { get; set; }

        [JsonPropertyName("item_sample")]
        public string? ItemSample { get; set; }

        [JsonPropertyName("amount")]
        public int Amount { get; set; } = 1;

        [JsonPropertyName("description")]
        public string? Description { get; set; }

        [JsonPropertyName("unit_price")]
        public decimal UnitPrice { get; set; }

        [JsonPropertyName("total_price")]
        public decimal TotalPrice { get; set; }
    }

    public class Quotation
    {
        [JsonPropertyName("id")]
        public Guid Id { get; set; }

        [JsonPropertyName("quotation_number")]
        public string QuotationNumber { get; set; } = string.Empty;

        [JsonPropertyName("year")]
        public int Year { get; set; }

        [JsonPropertyName("date")]
        public DateTime Date { get; set; }

        [JsonPropertyName("customer_id")]
        public Guid? CustomerId { get; set; }

        [JsonPropertyName("customer_order_id")]
        public Guid? CustomerOrderId { get; set; }

        [JsonPropertyName("client_name")]
        public string ClientName { get; set; } = string.Empty;

        [JsonPropertyName("client_ruc")]
        public string? ClientRuc { get; set; }

        [JsonPropertyName("client_direction")]
        public string? ClientDirection { get; set; }

        [JsonPropertyName("client_phone")]
        public string? ClientPhone { get; set; }

        [JsonPropertyName("client_email")]
        public string? ClientEmail { get; set; }

        [JsonPropertyName("project_responsable")]
        public string? ProjectResponsable { get; set; }

        [JsonPropertyName("project_ruc")]
        public string? ProjectRuc { get; set; }

        [JsonPropertyName("project_name")]
        public string? ProjectName { get; set; }

        [JsonPropertyName("project_location")]
        public string? ProjectLocation { get; set; }

        [JsonPropertyName("mobilization_cost")]
        public decimal MobilizationCost { get; set; }

        [JsonPropertyName("viatics_cost")]
        public decimal ViaticsCost { get; set; }

        [JsonPropertyName("subtotal_amount")]
        public decimal SubtotalAmount { get; set; }

        [JsonPropertyName("total_amount")]
        public decimal TotalAmount { get; set; }

        [JsonPropertyName("note")]
        public string? Note { get; set; }

        [JsonPropertyName("created_by")]
        public string? CreatedBy { get; set; }

        [JsonPropertyName("items")]
        public List<QuotationItem> Items { get; set; } = new();
    }

    public class QuotationItemCreate
    {
        [JsonPropertyName("item_norma")]
        public string? ItemNorma { get; set; }

        [JsonPropertyName("item_sample")]
        public string? ItemSample { get; set; }

        [JsonPropertyName("amount")]
        public int Amount { get; set; } = 1;

        [JsonPropertyName("description")]
        public string? Description { get; set; }

        [JsonPropertyName("unit_price")]
        public decimal UnitPrice { get; set; }
    }

    public class QuotationCreate
    {
        [JsonPropertyName("customer_id")]
        public Guid? CustomerId { get; set; }

        [JsonPropertyName("customer_order_id")]
        public Guid? CustomerOrderId { get; set; }

        [JsonPropertyName("client_name")]
        public string ClientName { get; set; } = string.Empty;

        [JsonPropertyName("client_ruc")]
        public string? ClientRuc { get; set; }

        [JsonPropertyName("client_direction")]
        public string? ClientDirection { get; set; }

        [JsonPropertyName("client_phone")]
        public string? ClientPhone { get; set; }

        [JsonPropertyName("client_email")]
        public string? ClientEmail { get; set; }

        [JsonPropertyName("project_responsable")]
        public string? ProjectResponsable { get; set; }

        [JsonPropertyName("project_ruc")]
        public string? ProjectRuc { get; set; }

        [JsonPropertyName("project_name")]
        public string? ProjectName { get; set; }

        [JsonPropertyName("project_location")]
        public string? ProjectLocation { get; set; }

        [JsonPropertyName("mobilization_cost")]
        public decimal MobilizationCost { get; set; }

        [JsonPropertyName("viatics_cost")]
        public decimal ViaticsCost { get; set; }

        [JsonPropertyName("note")]
        public string? Note { get; set; }

        [JsonPropertyName("items")]
        public List<QuotationItemCreate> Items { get; set; } = new();
    }
}

File: backend/app/schemas/quotation.py
from typing import Optional, List
from pydantic import BaseModel, computed_field
import datetime
import uuid

# QuotationItem Schemas
class QuotationItemBase(BaseModel):
    item_norma: Optional[str] = None
    item_sample: Optional[str] = None
    amount: int = 1
    description: Optional[str] = None
    unit_price: float = 0.0
    item_note: Optional[str] = None

class QuotationItemCreate(QuotationItemBase):
    pass

class QuotationItemUpdate(QuotationItemBase):
    pass

class QuotationItemInDBBase(QuotationItemBase):
    id: int
    quotation_id: uuid.UUID
    total_price: float

    model_config = {"from_attributes": True}

class QuotationItem(QuotationItemInDBBase):
    pass

# Quotation Schemas
class QuotationBase(BaseModel):
    client_name: str
    client_ruc: Optional[str] = None
    client_direction: Optional[str] = None
    client_phone: Optional[str] = None
    client_email: Optional[str] = None
    
    project_responsable: Optional[str] = None
    project_ruc: Optional[str] = None
    project_name: Optional[str] = None
    project_location: Optional[str] = None
    
    mobilization_cost: float = 0.0
    viatics_cost: float = 0.0
    note: Optional[str] = None

class QuotationCreate(QuotationBase):
    customer_id: Optional[uuid.UUID] = None
    customer_order_id: Optional[uuid.UUID] = None
    items: List[QuotationItemCreate] = []

class QuotationUpdate(QuotationBase):
    items: Optional[List[QuotationItemCreate]] = None

class QuotationInDBBase(QuotationBase):
    id: uuid.UUID
    quotation_number: str
    year: int
    date: datetime.date
    customer_order_id: Optional[uuid.UUID] = None
    created_by: Optional[str] = None
    subtotal_amount: float
    total_amount: float
    items: List[QuotationItem] = []

    model_config = {"from_attributes": True}

class Quotation(QuotationInDBBase):
    pass

class QuotationInDB(QuotationInDBBase):
    pass
